### PR TITLE
Add GraphQL mutations, general refactor

### PR DESCRIPTION
**Pull request by aitikgupta**
Added a separate interface for queries, handling different functionalities for projects.

### CODE REVIEW

File: webviews/components/ProjectInfo.svelte [hunk 1]
```diff
@@ -1,118 +1,166 @@
...
     type === "repo"
-      ? query(GET_REPO_PROJECT_INFO, {
+      ? query(queries.GET_REPO_PROJECT_INFO, {
           variables: {
             name: name,
             owner: owner,
             number: parseInt(number),
           },
-          pollInterval: 1800,
+          pollInterval: 100,
```
Comment (YuanSamuel) [inline webviews/components/ProjectInfo.svelte]: I think it's best to keep this a bit higher; GitHub has a rate limit, so we can't fetch data too often

**Comment by YuanSamuel** [comment]
Looks good atm, love how you're keeping the code clean

File: webviews/components/ProjectInfo.svelte (hunk 1 above)
Comment (aitikgupta) [reply webviews/components/ProjectInfo.svelte]: I was _just_ reading some blog that GitHub will ban you if you keep calling their APIs too often.
So +1, I'll edit it back :p

Comment (aitikgupta) [reply webviews/components/ProjectInfo.svelte]: It looks blazingly fast when we poll after every 100ms though

**Review by YuanSamuel — APPROVED** [review]
Looks good!

**Pull request merged**